android/src/test/java/cucumber/runtime/formatter/AndroidInstrumentationReporterTest.java

```diff
@@ -458,6 +464,116 @@ public void step_result_contains_only_the_current_scenarios_severest_result() {
+                formatter.finishTestCase();
+                // ... and finally after the test case has been finished.
+                testCaseFinished = formatter.getUniqueTestName(testCase);
+                return this;
+            }
+
+            private void assertValues(String expectedTestNameTestCaseStarted) {
+                assertThat("Unexpected test name on test case initialization", testCaseInitialized, is(nullValue()));
+                assertThat("Unexpected test name after test case start", testCaseStarted, equalTo(expectedTestNameTestCaseStarted));
+                assertThat("Unexpected test name after test step finished", testStepFinished, equalTo(expectedTestNameTestCaseStarted));
+                assertThat("Unexpected test name after test case finished",testCaseFinished, equalTo(expectedTestNameTestCaseStarted));
+            }
+        }
+        String[] featureNames = new String[]{"Addition", "Subtraction", "Multiplication", "Division"};
+        String[] scenarioNames = new String[] {"Enter one number", "Enter two numbers"};
+        List<TestNameTestSetup> testNameTestSetupCollection = new ArrayList<TestNameTestSetup>();
+        final AndroidInstrumentationReporter formatter = new AndroidInstrumentationReporter(runtime, instrumentation);
+        mockResultStatus(firstResult, Result.Type.PASSED);
+        final int MAX_EXAMPLES = 3;
+        // Normally, it is not good style to use for-loops in tests, but in this case it should be okay.
+        for (String featureName : featureNames) {
+            for (String scenarioName : scenarioNames) {
+                for (int exampleIndex = 0; exampleIndex < MAX_EXAMPLES; exampleIndex++) {
+                    TestNameTestSetup testNameTestSetup = new TestNameTestSetup(formatter, featureName, scenarioName);
+                    testNameTestSetup.invokeGetUniqueTestNameOnDifferentStatesInTestCaseLifecycle();
+                    testNameTestSetupCollection.add(testNameTestSetup);
+                }
+            }
+        }
+
+        testNameTestSetupCollection.add(new TestNameTestSetup(formatter, featureNames[0], scenarioNames[0])
+            .invokeGetUniqueTestNameOnDifferentStatesInTestCaseLifecycle()
+        );
+        testNameTestSetupCollection.add(new TestNameTestSetup(formatter, "New feature", scenarioNames[0])
+            .invokeGetUniqueTestNameOnDifferentStatesInTestCaseLifecycle()
+        );
+        testNameTestSetupCollection.add(new TestNameTestSetup(formatter, featureNames[0], "new_scenario")
+            .invokeGetUniqueTestNameOnDifferentStatesInTestCaseLifecycle()
+        );
+        testNameTestSetupCollection.add(new TestNameTestSetup(formatter, featureNames[0], "new_scenario")
+            .invokeGetUniqueTestNameOnDifferentStatesInTestCaseLifecycle()
+        );
+
+        // TODO Consider to split this test method into multiple ones, to fulfil the one-assert-per-test-method-paradigm
```

Comment:
Should this be done? (or remove "TODO"). In general, PR contains a several comments that should possible either be javadoc or be removed.

Reply:
@mlvandijk Thanks again for your feedback

I removed the TODO comment, added/improved comments and hopefully made the test method more readable